Dapr Traffic Control (#23490)
dotnet/docs: docs/architecture/dapr-for-net-developers/secrets-management.md

### FineCalculator component license key

The FineCollection service uses a component that calculates the fine based on the information of a speeding violation. This component is implemented as a domain service and is abstracted by the `IFineCalculator` interface:

```csharp
public interface IFineCalculator
{
    public int CalculateFine(string licenseKey, int violationInKmh);
}
```

The `CalculateFine` method expects a string containing a `licenseKey` as its first argument. This key unlocks the third-party component used by the implementation. To keep the example simple, the implementation hard-codes a series of `if` statements. You can find the implementation in the `HardCodedFineCalculator` class in the `DomainsServices` folder:

```csharp
    public class HardCodedFineCalculator : IFineCalculator
    {
        public int CalculateFine(string licenseKey, int violationInKmh)
        {
            if (licenseKey != "HX783-K2L7V-CRJ4A-5PN1G")
            {
                throw new InvalidOperationException("Invalid license-key specified.");
            }

>             int fine = 9; // default administration fee
>             if (violationInKmh < 5 )
>             {
>                 fine += 18;
>             }
>             else if (violationInKmh >= 5 && violationInKmh < 10 )
>             {
>                 fine += 31;
>             }
> 
>             // ...
>             
>             else if (violationInKmh == 35)
>             {
>                 fine += 372;
>             }
>             else
>             {
>                 // violation above 35 KMh will be determined by the prosecutor
>                 return 0;
>             }
> 
>             return fine;

Reviewer: ```suggestion
            int fine = 9; // default administration fee    
            return violationInKmh switch
            {
                < 5 => fine + 18,
                >= 5 and < 10 => fine + 31,

                // ...

                35 => fine + 372,
                // violation above 35 KMh will be determined by the prosecutor
                > 35 => 0,
                _ => fine
            };
```
